Diagnostics/HealthChecker/DataCollection/ExchangeInformation/Get-ExchangeServerCertificates.ps1

```diff
@@ -3,10 +3,84 @@
+            [bool]
+            $IsSelfSigned
+        )
+
+        [PSCustomObject]@{
+            IorSPattern  = $IssuerOrSubjectPattern
+            IsSelfSigned = $IsSelfSigned
+        }
+    }
+
+    function ShouldCertificateBeSkipped {
+        [OutputType("System.Boolean")]
+        param (
+            [Parameter(Mandatory = $true)]
+            [PSCustomObject]
+            $Exclusions,
+            [Parameter(Mandatory = $true)]
+            [System.Security.Cryptography.X509Certificates.X509Certificate2]
+            $Certificate
+        )
+
+        $certificateIndex = 0
+        $certificateMatch = $false
+
+        foreach ($exclusionEntry in $Exclusions) {
+            if ($certificateMatch -eq $false) {
+                switch ($exclusionEntry) {
+                    ({ $Certificate.Subject -match $_.IorSPattern }) { $certificateMatch = $true; break }
+                    ({ $Certificate.Issuer -match $_.IorSPattern }) { $certificateMatch = $true; break }
+                    default { $certificateIndex++ }
+                }
+            }
+        }
+
+        if ($certificateMatch) {
+            # We found a matching certificate based on subject or issuer, check for IsSelfSigned as second indicator
+            $skiplistEntry = $ExclusionObject[$certificateIndex]
+            if ($skiplistEntry -ne -1) {
+                return ($skiplistEntry.IsSelfSigned -eq $Certificate.IsSelfSigned)
+            } else {
+                return $false
+            }
+        } else {
+            return $false
+        }
+    }
+
     try {
+        if ($ExcludeCertificatesOnSkiplist) {
+            Write-Verbose "Build certificate exclusion list"
+            # Add the certificates that should be excluded from the Exchange certificate check (we don't return an object for them)
+            $certificatesToExclude = @(
+                # Exclude "MS-Organization-P2P-Access [YYYY]" certificate with one day lifetime on Azure hosted machines.
+                # See: What are the MS-Organization-P2P-Access certificates present on our Windows 10/11 devices?
+                # https://docs.microsoft.com/azure/active-directory/devices/faq
+                NewCertificateExclusionEntry -IssuerOrSubjectPattern "CN=MS-Organization-P2P-Access \[[12][0-9]{3}\]$" -IsSelfSigned $true
+                # Exclude "DC=Windows Azure CRP Certificate Generator" (TenantEncryptionCertificate)
+                # The certificates are built by the Azure fabric controller and passed to the Azure VM Agent.
+                # If you stop and start the VM every day, the fabric controller might create a new certificate.
+                # These certificates can be deleted. The Azure VM Agent re-creates certificates if needed.
+                # https://docs.microsoft.com/azure/virtual-machines/extensions/features-windows
+                NewCertificateExclusionEntry -IssuerOrSubjectPattern "DC=Windows Azure CRP Certificate Generator" -IsSelfSigned $true
```

Comment:
Let's clean up the comments to be outside the list.
```suggestion
            # Add the certificates that should be excluded from the Exchange certificate check (we don't return an object for them)
                # Exclude "MS-Organization-P2P-Access [YYYY]" certificate with one day lifetime on Azure hosted machines.
                # See: What are the MS-Organization-P2P-Access certificates present on our Windows 10/11 devices?
                # https://docs.microsoft.com/azure/active-directory/devices/faq
                 # Exclude "DC=Windows Azure CRP Certificate Generator" (TenantEncryptionCertificate)
                # The certificates are built by the Azure fabric controller and passed to the Azure VM Agent.
                # If you stop and start the VM every day, the fabric controller might create a new certificate.
                # These certificates can be deleted. The Azure VM Agent re-creates certificates if needed.
                # https://docs.microsoft.com/azure/virtual-machines/extensions/features-windows
            $certificatesToExclude = @(
                NewCertificateExclusionEntry  "CN=MS-Organization-P2P-Access \[[12][0-9]{3}\]$" $true
                NewCertificateExclusionEntry "DC=Windows Azure CRP Certificate Generator" $true
```
You should be able to get by with two parameters without needing to pass the parameter name.